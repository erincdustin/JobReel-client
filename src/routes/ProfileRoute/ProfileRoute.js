import React from 'react';
import SideNav from '../../components/SideNav/SideNav';
import UserProfile from '../../components/UserProfile/UserProfile';
import EditProfileForm from '../../components/EditProfileForm/EditProfileForm';
import jobReelContext from '../../context/JobReelContext';
import jobReelApiService from '../../services/jobreel-api-service';
import './ProfileRoute.css'

class ProfileRoute extends React.Component {
  
  constructor() {
    super();
    this.state = {
      user: {},
      editingProfile: false
    }
  }
  
  static contextType = jobReelContext;

  componentDidMount() {
    jobReelApiService.getUserById(this.context.user.id)
      .then(res => {
        this.setUser(res);
      })
  }

  setUser = (user) => {
    user.id = this.context.user.id;
    this.setState({ user })
  }

  updateEditingProfile = () => {
    this.setState({ editingProfile: !this.state.editingProfile });
  }

  render() {
    let display;
    (this.state.editingProfile) ? display = <EditProfileForm user={this.state.user} setUser={this.setUser} updateEditingProfile={this.updateEditingProfile}/> : display = <UserProfile user={this.state.user} updateEditingProfile={this.updateEditingProfile}/>
    return (
      <div className="ProfileRoute">
        <div className='title'>
          <h2>Profile</h2>
        </div>
        <SideNav />
        {display}
      </div>
    )
  }
}

export default ProfileRoute;