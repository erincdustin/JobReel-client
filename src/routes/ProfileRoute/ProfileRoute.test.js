import React from 'react'
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import ProfileRoute from './ProfileRoute'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PropTypes from 'prop-types';

class ContextProvider extends React.Component {
  static childContextTypes = {
    myContext: PropTypes.object,
  }

  getChildContext = () => ({
    myContext: {
      user: {
        id: 1
      }
    }
  })

  render() {
    return this.props.children;
  }
}

describe('ProfileRoute smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><ProfileRoute /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('ProfileRoute snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(
      <ContextProvider>
        <ProfileRoute />
      </ContextProvider>
    )
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})