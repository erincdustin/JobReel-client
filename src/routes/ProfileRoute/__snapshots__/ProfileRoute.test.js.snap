// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileRoute snapshot test renders the UI as expected 1`] = `<ProfileRoute />`;
